src/coprocessor_v2/plugin_registry.rs

```diff
@@ -307,18 +302,18 @@ impl PluginRegistryInner {
     pub fn load_plugin<P: AsRef<OsStr>>(
         &mut self,
         filename: P,
-    ) -> Result<&'static str, PluginLoadingError> {
+    ) -> Result<String, PluginLoadingError> {
```

Comment:
I think we could in theory still return `&'static str` if we store `name: &'static str` in `LoadedPlugin` instead of `name: String`.
But I'm not sure what is nicer? Any suggestions?

Reply:
We would have fewer constraints on lifetime if we use `String`, and I suggest doing so.